clang/lib/Sema/SemaOverload.cpp

```diff
@@ -4404,14 +4404,19 @@ CompareImplicitConversionSequences(Sema &S, SourceLocation Loc,
     Result = CompareStandardConversionSequences(S, Loc,
                                                 ICS1.Standard, ICS2.Standard);
   else if (ICS1.isUserDefined()) {
+    auto ConvFunc1 = ICS1.UserDefined.ConversionFunction;
```

Comment:
```suggestion
    const FunctionDecl* ConvFunc1 = ICS1.UserDefined.ConversionFunction;
```

Reply:
The LLVM Coding Standards argue for `auto` in "other places where the type is already obvious from the context." I thought that's the case here...

Reply:
> The LLVM Coding Standards argue for `auto` in "other places where the type is already obvious from the context." I thought that's the case here...

That mostly applies to cases where the type is already explicitly mentioned in the initialiser, usually because it’s some sort of `cast<>`.